frontend/src/components/common/Pagination.tsx

```diff
@@ -0,0 +1,134 @@
+import React from "react";
```

Comment:
super duper happy that this is its own component :) we can (hopefully) re-use it with a little bit of tweaking for the other pages